src/schemas.js

```diff
@@ -171,3 +170,247 @@ export const atomicSwapCreateSchema = {
+    in: ['body'],
+    errorMessage: 'Invalid inputs array',
+    isArray: true,
+    isLength: 1,
+    notEmpty: true,
+  },
+  'inputs.*.type': {
+    in: ['body'],
+    errorMessage: 'Invalid type',
+    isString: true,
+    equals: {
+      options: 'query',
+    },
+  },
+  'inputs.*.max_utxos': {
+    in: ['body'],
+    errorMessage: 'Invalid max_utxos',
+    isInt: {
+      options: {
+        min: 1,
+      },
+    },
+    toInt: true,
+    optional: true,
+  },
+  'inputs.*.filter_address': {
+    in: ['body'],
+    errorMessage: 'Invalid filter_address',
+    isString: true,
+    optional: true,
+  },
+  'inputs.*.amount_smaller_than': {
+    in: ['body'],
+    errorMessage: 'Invalid amount_smaller_than',
+    isInt: {
+      options: {
+        min: 2,
+      },
+    },
+    toInt: true,
+    optional: true,
+  },
+  'inputs.*.amount_bigger_than': {
+    in: ['body'],
+    errorMessage: 'Invalid amount_bigger_than',
+    isInt: {
+      options: {
+        min: 1,
+      },
+    },
+    toInt: true,
+    optional: true,
+  },
+  'inputs.*.maximum_amount': {
+    in: ['body'],
+    errorMessage: 'Invalid maximum_amount',
+    isInt: {
+      options: {
+        min: 1,
+      },
+    },
+    toInt: true,
+    optional: true,
+  },
+};
+
+export const txHexInputDataSchema = {
+  ...txHexSchema,
+  signatures: {
+    in: ['body'],
+    errorMessage: 'Invalid signatures',
+    isArray: true,
+    isEmpty: false,
+  },
+  'signatures.*.index': {
+    in: ['body'],
+    errorMessage: 'Invalid signature input index',
+    isInt: {
+      options: {
+        min: 0,
+      },
+    },
+    toInt: true,
+  },
+  'signatures.*.data': {
+    in: ['body'],
+    errorMessage: 'Invalid signature data',
+    isString: true,
+    custom: {
+      options: value => /^[0-9a-fA-F]+$/.test(value),
+    },
+  },
+};
+
+export const p2pkhSignature = {
+  index: {
+    in: ['body'],
+    errorMessage: 'Invalid index',
+    isInt: {
+      options: {
+        min: 0,
+      },
+    },
+    toInt: true,
+  },
+  signature: {
+    in: ['body'],
+    errorMessage: 'Invalid signatures',
+    isString: true,
+    custom: {
+      options: value => /^[0-9a-fA-F]+$/.test(value)
+    },
+  },
+  type: {
+    customSanitizer: {
+      options: () => 'p2pkh',
+    }
+  },
+};
+
+export const p2shSignature = {
+  index: {
+    in: ['body'],
+    errorMessage: 'Invalid index',
+    isInt: {
+      options: {
+        min: 0,
+      },
+    },
+    toInt: true,
+  },
+  signatures: {
+    in: ['body'],
+    errorMessage: 'Invalid signature',
+    isObject: true,
+    custom: {
+      options: value => {
+        for (const [xpub, signature] of Object.entries(value)) {
+          // values should be hex strings and keys should be valid xpubs
+          if (!(/^[0-9a-fA-F]+$/.test(signature)
+            && walletUtils.isXpubKeyValid(xpub))) return false;
+        }
+        return true;
+      }
+    },
+  },
+  type: {
+    customSanitizer: {
+      options: () => 'p2sh',
```

Comment:
What's this `p2sh` used for?

Reply:
This is not an argument, the sanitizer will add the `type: "p2sh"` or `type: "p2pkh"` depending on which validation chain is resolved for the `oneOf`.

This way we can have different arguments based on the `p2sh` or `p2pkh` and this `type` will indicate for the controller which one it is.